### ISSUE TITLE
Trying to create a second ERB tag adjusts the first tag's type.

### ISSUE THREAD

**Issue by TimSullivan** [devjah/rails-nova#1]
Hi there! 

When using the plugin, pressing SHIFT-CMD-. a tag is created only if there's not already an ERB tag on the line. If an existing tag is there, even if I'm not within it, it toggles through the available ERB styles rather than creating a new tag.

Similarly, if I manually type the second tag, the shortcut only affects the first tag on the line. 

![Example](https://user-images.githubusercontent.com/16880/109177381-bce54680-7755-11eb-96d5-7397502fbf91.gif)

**Comment by devjah** [comment]
I was aware of this bug, but I stopped using Nova temporarily until they fix a few bugs that annoyed me.
I'm glad there are people that use this extension and even more glad that somebody took the time to report an issue.
Thank you, Tim!
Just pushed an update that fixes this.
❤️

**Comment by TimSullivan** [comment]
Thanks for taking a look! Though now the shortcut just stopped working most of the time. The first time I tried it, nothing happened, so I restarted Nova, tried it again, and it worked. Moved to a new line, tried again, didn't work. Deleted the original `<%= %>` and tried again, it worked. I can't notice any pattern to when it works and when it doesn't, though.

To duplicate, I went to the end of my code, on a new line entered this (the | is the insertion point!):

    text |

then used the keyboard shortcut and it worked. Added a command into the ERB, moved outside it:

    text <%= something %> |

...and the shortcut didn't work.

The console log:

Ruby on Rails ERB tags[10:37:26.176000] ————————————————————————————————————————————

Ruby on Rails ERB tags[10:37:26.177000] 🎉

Ruby on Rails ERB tags[10:37:26.177000] hasSelection: false

Ruby on Rails ERB tags[10:37:26.177000] selectedRanges: [10748, 10748]

Ruby on Rails ERB tags[10:37:26.177000] entireLineRanges: [10726, 10749]

Ruby on Rails ERB tags[10:37:26.177000] linePositions: 22

Ruby on Rails ERB tags[10:37:26.177000] 🎉

Ruby on Rails ERB tags[10:37:26.177000] [0, 10748]

Ruby on Rails ERB tags[10:37:26.178000] <% occurence 154

Ruby on Rails ERB tags[10:37:26.178000] %> occurence 31

Ruby on Rails ERB tags[10:37:26.178000] No selection

Ruby on Rails ERB tags[10:37:26.178000] It's inside another tag

Ruby on Rails ERB tags[10:37:26.178000] lastOpeningTagBeforeCursor: <%=

Ruby on Rails ERB tags[10:37:26.178000] changing bracket 1

Ruby on Rails ERB tags[10:37:26.178000] bracketsToUseIndex: 1

Ruby on Rails ERB tags[10:37:26.178000] editor range: [10748, 10748]

[... removed code ...]
text <%= something %> 

Ruby on Rails ERB tags[10:37:26.178000] fromLastagToCursor: undefined

Ruby on Rails ERB tags[10:37:26.178000] ERB tags error

**Comment by devjah** [comment]
```
Ruby on Rails ERB tags[10:37:26.178000] <% occurence 154
Ruby on Rails ERB tags[10:37:26.178000] %> occurence 31
```
According to these lines, you have 154 opening tags before the cursor and 31 closing
The number should be either equal which would mean that the cursor is outside any ERB tags
Or if there are more opening tags (usually 1 more) that would mean the cursor is inside a tag.

I'm wondering how is it possible that you have 123 more opening tags than closing (before the cursor)... Do you have a string somewhere that includes combinations of `<%`?

**Comment by TimSullivan** [comment]
Just checked, and nope. 31 of both opening and closing. The view renders, so there's no mismatched tags that I can see. It does seem to happen more often in larger, more complex files, for sure, but none of them have more <% than %>, so that's super weird.

In a file where it doesn't seem to work anywhere, if I add blank lines to the top, it seems to work there, but not within the file. Here's code I can confirm has issues:

```
<div class="mt-5 bg-red-50 md:rounded-xl">  
  <div class="max-w-7xl mx-auto py-12 px-4 sm:px-6 lg:py-18 lg:px-8 lg:flex lg:items-center lg:justify-between">
    <h2 class="text-3xl font-extrabold tracking-tight text-gray-900 md:text-4xl">
      <span class="block">Text</span>
      <span class="block text-red-600">text</span>
    </h2>
    <div class="mt-8 flex lg:mt-0 lg:flex-shrink-0">
      <div class="inline-flex rounded-md shadow">
        <%= link_to "Text.", some_path, class: 'btn btn-primary text-lg' %>
        
      </div>
    </div>
  </div>
  <%= something %>
  
</div>
```

If you try to use the shortcut after the `<%= something %>` tag, it doesn't work.

**Comment by devjah** [comment]
Indeed, there was an issue with the regex that detects the opening tags.
Should be fixed in 1.1.1
Thanks for the snippet, Tim!

the regex was like this `/<%?./g` instead of this `/<%.?/g`
so basically all HTML tag openings were counting in :)
